pkg/exec/error_test.go

```diff
@@ -0,0 +1,120 @@
+package exec_test
+
+import (
+	"context"
+	"fmt"
+	"github.com/applike/gosoline/pkg/exec"
+	"github.com/hashicorp/go-multierror"
+	"github.com/stretchr/testify/assert"
+	"io"
+	"net"
+	"testing"
+)
+
+func TestIsRequestCanceled(t *testing.T) {
+	for name, test := range map[string]struct {
+		err        error
+		isCanceled bool
+	}{
+		"none": {
+			err:        nil,
+			isCanceled: false,
+		},
+		"other error": {
+			err:        io.EOF,
+			isCanceled: false,
+		},
+		"format error": {
+			err:        fmt.Errorf("error: %d", 42),
+			isCanceled: false,
+		},
+		"simple": {
+			err:        context.Canceled,
+			isCanceled: true,
+		},
+		"simple wrapped": {
+			err:        fmt.Errorf("error %w", context.Canceled),
+			isCanceled: true,
+		},
+		"exec": {
+			err:        exec.RequestCanceledError,
+			isCanceled: true,
+		},
+		"exec wrapped": {
+			err:        fmt.Errorf("error %w", exec.RequestCanceledError),
+			isCanceled: true,
+		},
+		"multierror empty": {
+			err:        multierror.Append(nil),
+			isCanceled: false,
+		},
+		"multierror single": {
+			err:        multierror.Append(nil, context.Canceled),
+			isCanceled: true,
+		},
+		"multierror single wrapped": {
+			err:        multierror.Append(nil, fmt.Errorf("error %w", context.Canceled)),
+			isCanceled: true,
+		},
+		"multierror multiple wrapped": {
+			err:        multierror.Append(nil, fmt.Errorf("error %w", context.Canceled), fmt.Errorf("error %w", exec.RequestCanceledError)),
+			isCanceled: true,
+		},
+		"multierror mixed": {
+			err:        multierror.Append(nil, context.Canceled, io.EOF),
+			isCanceled: false,
+		},
+	} {
+		test := test
+		t.Run(name, func(t *testing.T) {
+			assert.Equal(t, test.isCanceled, exec.IsRequestCanceled(test.err))
+		})
+	}
+}
+
+func TestIsConnectionResetError(t *testing.T) {
+	var err error
+	var conn net.Conn
+	var listener net.Listener
+	var waitListen = make(chan struct{})
+
+	go func() {
+		var err error
+		var conn net.Conn
+
+		if listener, err = net.Listen("tcp", "localhost:0"); err != nil {
+			assert.FailNow(t, err.Error(), "can not create listener")
+			return
+		}
+
+		close(waitListen)
+
+		if conn, err = listener.Accept(); err != nil {
+			assert.FailNow(t, err.Error(), "can not accept connection")
```

Comment:
`/usr/lib/go/src/testing/testing.go:633` writes:

// A test ends when its Test function returns or calls any of the methods
// FailNow, Fatal, Fatalf, SkipNow, Skip, or Skipf. Those methods, as well as
// the Parallel method, must be called only from the goroutine running the
// Test function.

This is most likely fine (some implementations just do `exit(1)`), but in theory this code is not obeying these rules.